### PR TITLE
Us84

### PR DESCRIPTION
**Pull request by frcroth**
#84

### CODE REVIEW

**Comment by sirkrypt0** [comment]
I tested it locally on Linux, loading a file works very well.
But as soon as the file is loaded into the editor, applying any structure to that text takes years. Opening a new editor and pasting the loaded text in the new one works and applying structures to the text in the new editor is fast again.
Not sure why this happens

Steps to reproduce:

- Open a SquidWord (1)
- Load a file
- Try to apply structures to that text (takes really long)
- Open a new SquirdWord (2) (leaving the other one open)
- Copy the text from (1) into (2)
- Apply structures to the text in (2) (is fast again)

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/loadFileFrom..st [hunk 1]
```diff
@@ -0,0 +1,8 @@
+saving and loading
+loadFileFrom: filePath
+
+	|filestream text document|
```
Comment (sirkrypt0) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/loadFileFrom..st]: Have we decided on spaces before and after local variables yet?

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/loadFileFrom..st [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+	filestream := StandardFileStream readOnlyFileNamed: filePath.
```
Comment (sirkrypt0) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/loadFileFrom..st:5]: Not sure if there are any best practices on that, but I think we should close the `filestream` after reading

File: packages/RichTextEditing-Tests.package/RichTextEditorFileManipulationTests.class/instance/testUS84loadFileFromFilesystem.st [hunk 3]
```diff
@@ -0,0 +1,6 @@
+acceptance-tests
+testUS84loadFileFromFilesystem
```
Comment (sirkrypt0) [inline packages/RichTextEditing-Tests.package/RichTextEditorFileManipulationTests.class/instance/testUS84loadFileFromFilesystem.st:2]: Missing newline after method name

File: packages/RichTextEditing-Tests.package/RichTextEditorFileManipulationTests.class/instance/testUS84loadFileFromFilesystem.st [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+	self setupFile.
```
Comment (sirkrypt0) [inline packages/RichTextEditing-Tests.package/RichTextEditorFileManipulationTests.class/instance/testUS84loadFileFromFilesystem.st]: We could also use the generic `setUp` and `tearDown` methods and call this there and thus not having to do setup and teardown in the testcase itself

**Review by sirkrypt0 — CHANGES_REQUESTED** [review]

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/loadFileFrom..st (hunk 2 above)
Comment (sirkrypt0) [reply packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/loadFileFrom..st:5]: fixed that

File: packages/RichTextEditing-Tests.package/RichTextEditorFileManipulationTests.class/instance/testUS84loadFileFromFilesystem.st (hunk 3 above)
Comment (sirkrypt0) [reply packages/RichTextEditing-Tests.package/RichTextEditorFileManipulationTests.class/instance/testUS84loadFileFromFilesystem.st:2]: fixed that

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/loadFileFrom..st (hunk 1 above)
Comment (sirkrypt0) [reply packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/loadFileFrom..st]: fixed that

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/loadFileFrom..st (hunk 2 above)
Comment (frcroth) [reply packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/loadFileFrom..st:5]: Yes definitely. Done in the tests, so I can easily implement this

**Comment by frcroth** [comment]
> 
> 
> I tested it locally on Linux, loading a file works very well.
> But as soon as the file is loaded into the editor, applying any structure to that text takes years. Opening a new editor and pasting the loaded text in the new one works and applying structures to the text in the new editor is fast again.
> Not sure why this happens
> 
> Steps to reproduce:
> 
>     * Open a SquidWord (1)
> 
>     * Load a file
> 
>     * Try to apply structures to that text (takes really long)
> 
>     * Open a new SquirdWord (2) (leaving the other one open)
> 
>     * Copy the text from (1) into (2)
> 
>     * Apply structures to the text in (2) (is fast again)

Can't reproduce this on my machine (Windows)

**Comment by sirkrypt0** [comment]
> > I tested it locally on Linux, loading a file works very well.
> > But as soon as the file is loaded into the editor, applying any structure to that text takes years. Opening a new editor and pasting the loaded text in the new one works and applying structures to the text in the new editor is fast again.
> > Not sure why this happens
> > Steps to reproduce:
> > ```
> > * Open a SquidWord (1)
> > 
> > * Load a file
> > 
> > * Try to apply structures to that text (takes really long)
> > 
> > * Open a new SquirdWord (2) (leaving the other one open)
> > 
> > * Copy the text from (1) into (2)
> > 
> > * Apply structures to the text in (2) (is fast again)
> > ```
> 
> Can't reproduce this on my machine (Windows)

Hm crazy, lets wait what @oleschl says

**Comment by oleschl** [comment]
> > > I tested it locally on Linux, loading a file works very well.
> > > But as soon as the file is loaded into the editor, applying any structure to that text takes years. Opening a new editor and pasting the loaded text in the new one works and applying structures to the text in the new editor is fast again.
> > > Not sure why this happens
> > > Steps to reproduce:
> > > ```
> > > * Open a SquidWord (1)
> > > 
> > > * Load a file
> > > 
> > > * Try to apply structures to that text (takes really long)
> > > 
> > > * Open a new SquirdWord (2) (leaving the other one open)
> > > 
> > > * Copy the text from (1) into (2)
> > > 
> > > * Apply structures to the text in (2) (is fast again)
> > > ```
> > 
> > 
> > Can't reproduce this on my machine (Windows)
> 
> Hm crazy, lets wait what @oleschl says

I sometimes have similar issues as @sirkrypt0. However, it might not be a performance issue but a reloading issue. @sirkrypt0 can you load file, select some text, apply a structure with the mouse and move the cursor into the text field again? 

Also when using the structure shortcuts, the structures are applied instantly.

**Comment by sirkrypt0** [comment]
> > > > I tested it locally on Linux, loading a file works very well.
> > > > But as soon as the file is loaded into the editor, applying any structure to that text takes years. Opening a new editor and pasting the loaded text in the new one works and applying structures to the text in the new editor is fast again.
> > > > Not sure why this happens
> > > > Steps to reproduce:
> > > > ```
> > > > * Open a SquidWord (1)
> > > > 
> > > > * Load a file
> > > > 
> > > > * Try to apply structures to that text (takes really long)
> > > > 
> > > > * Open a new SquirdWord (2) (leaving the other one open)
> > > > 
> > > > * Copy the text from (1) into (2)
> > > > 
> > > > * Apply structures to the text in (2) (is fast again)
> > > > ```
> > > 
> > > 
> > > Can't reproduce this on my machine (Windows)
> > 
> > 
> > Hm crazy, lets wait what @oleschl says
> 
> I sometimes have similar issues as @sirkrypt0. However, it might not be a performance issue but a reloading issue. @sirkrypt0 can you load file, select some text, apply a structure with the mouse and move the cursor into the text field again?
> 
> Also when using the structure shortcuts, the structures are applied instantly.

This actually works

**Review by sirkrypt0 — APPROVED** [review]

**Review by oleschl — APPROVED** [review]

**Pull request merged**